src/server/index.ts

```diff
@@ -0,0 +1,86 @@
+import webpack from 'webpack';
+import path from 'path';
+import {createWebpackConfiguration} from '../webpack.config';
+import WebpackDevServer from 'webpack-dev-server';
+
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const {WebWorkerPlugin} = require('@shopify/web-worker/webpack');
+
+interface ServerConfig {
+  port: number;
+  entry: string;
+}
+
+export async function server({port, entry}: ServerConfig) {
+  const url = `http://localhost:${port}`;
+
+  const staticCompiler = webpack(
+    createWebpackConfiguration({
+      mode: 'development',
+      target: 'webworker',
+      entry: path.resolve(entry),
+      output: {
+        globalObject: 'self',
+        filename: 'third-party-script.js',
+        path: path.resolve(__dirname, 'build'),
+        publicPath: '/build/',
+      },
+      devtool: 'source-map',
+    })
+  );
+  staticCompiler.run((err) => {
+    if (err) {
+      console.error('Error compiling', err);
+    }
+  });
+
+  const serverCompiler = webpack([
+    createWebpackConfiguration({
+      mode: 'development',
+      target: 'web',
+      entry: {
+        main: path.resolve(__dirname, './host/index'),
+      },
+      output: {
+        globalObject: 'self',
+        filename: '[name].js',
+        path: __dirname + '/build',
+        publicPath: `${url}/`,
+      },
+      plugins: [
+        new WebWorkerPlugin(),
+        new HtmlWebpackPlugin(),
+        new webpack.DefinePlugin({
+          THIRD_PARTY_SCRIPT: `"${url}/build/third-party-script.js"`,
+        }),
+      ],
+      devtool: 'source-map',
```

Comment:
Do we need a source map for the host? I don't think developers should be able to debug the host code

Reply:
True